src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> elements = null;
+        if (type == Truck.class) {
+            elements = new TruckProducer().get();
+        } else if (type == Excavator.class) {
+            elements = new ExcavatorProducer().get();
+        } else if (type == Bulldozer.class) {
+            elements = new BulldozerProducer().get();
```

Comment:
producers must be class-level fields and created once, not every time this method being called

Reply:
++